src/WebJobs.Script/Workers/Rpc/RpcWorkerChannel.cs

```diff
@@ -211,7 +211,7 @@ internal void FunctionEnvironmentReloadResponse(FunctionEnvironmentReloadRespons
             if (res.Result.IsFailure(out Exception reloadEnvironmentVariablesException))
             {
                 _workerChannelLogger.LogError(reloadEnvironmentVariablesException, "Failed to reload environment variables");
-                _reloadTask.SetResult(false);
+                _reloadTask.SetException(reloadEnvironmentVariablesException);
```

Comment:
is something else catching this exception?

Reply:
yes. Setting this exception will set the task to status to faulted which is handled here:https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/Standby/StandbyManager.cs#L78